docs/adr/0014-auth-for-settings-redesign.md

```diff
@@ -0,0 +1,87 @@
+# Authentication in Settings Redesign
+
+- Status: proposed
+- Deciders: Les Orchard
+- Date: 2020-04-14
+
+Issue: https://jira.mozilla.com/browse/FXA-1559
+
+## Context and Problem Statement
+
+The Settings Redesign app needs to query & mutate the same protected user data as the existing settings app hosted on content-server. This will require some form of authentication & authorization to manage that data.
+
+## Decision Drivers
+
+- Smooth UX
+- Security
+- Development velocity
+- Ease of integration
+
+## Considered Options
+
+- Share existing sessionToken to use auth-server APIs
```

Comment:
So for my own understanding here, the payments server uses an access token. I believe we use the session token to generate the access token, or I at least know we need to generate an oauth access token for the user's profile from the session token, so how does this differ from payments/why is it better? 🤔

Reply:
The payment-server app and the auth-server APIs for subscriptions were built from the ground up to work with OAuth access tokens. 

In contrast, the content-server and auth-server APIs for settings management generally do *not* accept OAuth tokens, only the session token from FxA login.

We pass payments-server a pre-generated OAuth token via a redirect from content-server. And, yeah, that requires the user to have a session token from an active FxA login.

But, this is a hack to smooth over the OAuth UX flow for payments-server. [We've been meaning to fix it since last July](https://github.com/mozilla/fxa/issues/1695). So, I wouldn't want to apply that particular hack to a second project *and* need to modify most of auth-server to accept an OAuth token alongside session token.

That said, I am proposing another kind of hack to convey the session token to the new settings redesign app.

But, this hack at least doesn't require auth-server changes. And, it could eventually be un-hacked if/when the settings redesign is hosted from content-server - i.e. because the app would have normal access to the session token from the proper accounts.firefox.com origin

Reply:
Oh yeah and another reason payments-server uses OAuth access token rather than the session token: I seem to remember we had a notion that Relying Parties beyond payment-server might call subscription management APIs. OAuth has scopes that can be attached to tokens to control what APIs the token can be used with.

The session token, on the other hand, is like root sudo access - there is no granular access control on the session token. But, it's only ever used with our own settings management app.

I could someday imagine us tweaking settings management APIs to work with scoped OAuth access tokens. But, that doesn't seem to me like critical path work for the settings redesign.

Reply:
Thanks for the explainer. I thought it might be something along the lines of "payments only needs an access token but settings requires the entire session token," but it makes a lot more sense with all that context. 👍

Reply:
👀 Bookmarking this.